src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp

```diff
@@ -101,6 +116,158 @@ class PolicyManager final : public Singleton<PolicyManager>
+     * required fields and has valid values. If the validation passes, no exception is thrown.
+     *
+     * @param vdObj A constant reference to a JSON object representing the vulnerability detection's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throws std::runtime exception.
+     */
+
+    void validateAndConfigureVulnerabilityDetection(const nlohmann::json& vdObj)
+    {
+        bool isValidUrl {false};
+
+        if (!vdObj.contains("enabled"))
+        {
+            throw std::runtime_error("Missing enabled field.");
+        }
+
+        if (!vdObj.contains("index-status"))
+        {
+            throw std::runtime_error("Missing index-status field.");
+        }
+
+        if (vdObj.contains("feed-update-interval"))
+        {
+            m_feedUpdateTime = Utils::parseStrToTime(vdObj["feed-update-interval"]) != -1
+                                   ? Utils::parseStrToTime(vdObj["feed-update-interval"])
+                                   : m_feedUpdateTime;
+        }
+
+        if (vdObj.contains("offline-url"))
+        {
+            isValidUrl = Utils::startsWith(vdObj["offline-url"], "file") ||
+                         Utils::startsWith(vdObj["offline-url"], "http") ||
+                         Utils::startsWith(vdObj["offline-url"], "https");
+        }
+
+        try
+        {
+            m_vulDetectEnabled = m_vulDetectEnabled && Utils::parseStrToBool(vdObj["enabled"]);
+
+            m_indexerEnabled = m_indexerEnabled && Utils::parseStrToBool(vdObj["index-status"]);
+        }
+        catch (const std::exception& e)
+        {
+            return;
+        }
+
+        if (isValidUrl)
+        {
+            m_urlFeed = std::move(vdObj["offline-url"]);
+        }
+    }
+
+    /**
+     * @brief Validates and configures the indexer based on the provided JSON object.
+     *
+     * This function takes a JSON object as input, which is expected to contain configuration
+     * information for the indexer. It validates the JSON object to ensure it contains the
+     * required fields and has valid values, based on the previous configuration of the vulnerability detection.
+     * If the validation passes, no exception is thrown.
+     *
+     * @param idObj A constant reference to a JSON object representing the indexer's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throws std::runtime exception.
+     */
+
+    void validateAndConfigureIndexer(const nlohmann::json& idObj)
+    {
+        if (!idObj.contains("enabled"))
+        {
+            throw std::runtime_error("Missing enabled field.");
+        }
+
+        if (!(Utils::parseStrToBool(idObj["enabled"]) && m_indexerEnabled))
```

Comment:
I'm not understanding this. If m_indexerEnabled is true (by default) and idObj["enabled"] is "no" it will return an "invalid configuration" exception. Is that ok?

Reply:
Good catch, the true table is:

|indexer status| indexer in vulnerability-detection | Result |
|---------------------|-------------------------------------------------|-----------|
| `true` | `true` | ✔️  It's valid |
| `false` | `true` | ❌ Not valid, throw exception  |
| `true` | `false` | ❌ Not valid, throw exception  |
| `false` | `false` | ❌ Not valid, throw exception |